### PR TITLE
Feature/message list view

### PR DESCRIPTION
**Pull request by adrian09h**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemAdapter.java [hunk 1]
```diff
@@ -82,8 +82,9 @@ public void replaceEntities(List<MessageListItem> newEntities) {
                 new MessageListItemDiffCallback(messageListItemList, newEntities), true);
 
         // only update those rows that change...
-        result.dispatchUpdatesTo(this);
+
         messageListItemList = newEntities;
+        result.dispatchUpdatesTo(this);
     }
```
Comment (tschellenbach) [inline library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemAdapter.java]: isn't it better to first dispatchUpdates and after that swap the variable?

Comment (adrian09h) [reply library/src/main/java/com/getstream/sdk/chat/adapter/MessageListItemAdapter.java]: Yep, that is the normal, but not important.

File: sample/src/main/res/layout/activity_channel.xml [hunk 2]
```diff
@@ -11,31 +10,25 @@
             type="com.getstream.sdk.chat.viewmodel.ChannelViewModel" />
     </data>
 
-    <androidx.constraintlayout.widget.ConstraintLayout
+    <RelativeLayout
```
Comment (tschellenbach) [inline sample/src/main/res/layout/activity_channel.xml]: it's recommended to always use constraint layout these days right?

File: library/src/main/java/com/getstream/sdk/chat/view/ChannelHeaderView.java [hunk 3]
```diff
@@ -52,9 +52,9 @@ public void setViewModel(ChannelViewModel model, LifecycleOwner lifecycleOwner)
         configHeaderAvatar();
     }
 
-    private ToolbarChannelHeaderBinding initBinding(Context context) {
+    private ViewChannelHeaderBinding initBinding(Context context) {
         LayoutInflater inflater = LayoutInflater.from(context);
-        binding = ToolbarChannelHeaderBinding.inflate(inflater, this, true);
+        binding = ViewChannelHeaderBinding.inflate(inflater, this, true);
```
Comment (tschellenbach) [inline library/src/main/java/com/getstream/sdk/chat/view/ChannelHeaderView.java]: good rename

Comment (adrian09h) [reply library/src/main/java/com/getstream/sdk/chat/view/ChannelHeaderView.java]: :)

**Pull request merged**